atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -128,28 +107,21 @@ public LockRefreshToken apply(HeldLocksToken input) {
     }
 
     public RawTransaction setupRunTaskWithLocksThrowOnConflict(Iterable<LockRefreshToken> lockTokens) {
-        long immutableLockTs = timestampService.getFreshTimestamp();
-        Supplier<Long> startTimestampSupplier = getStartTimestampSupplier();
-        LockDescriptor lockDesc = AtlasTimestampLockDescriptor.of(immutableLockTs);
-        LockRequest lockRequest = LockRequest.builder(ImmutableSortedMap.of(lockDesc, LockMode.READ))
-                .withLockedInVersionId(immutableLockTs).build();
-        LockRefreshToken lock;
+        LockImmutableTimestampResponse immutableTsResponse = timelockService.lockImmutableTimestamp(
+                LockImmutableTimestampRequest.create());
         try {
-            lock = lockService.lock(lockClient.getClientId(), lockRequest);
-        } catch (InterruptedException e) {
-            throw Throwables.throwUncheckedException(e);
-        }
-        try {
-            ImmutableList<LockRefreshToken> allTokens = ImmutableList.<LockRefreshToken>builder()
-                    .add(lock)
-                    .addAll(lockTokens)
-                    .build();
-            SnapshotTransaction transaction = createTransaction(immutableLockTs, startTimestampSupplier, allTokens);
-            return new RawTransaction(transaction, lock);
+            LockTokenV2 immutableTsLock = immutableTsResponse.getLock();
+            long immutableTs = immutableTsResponse.getImmutableTimestamp();
+            recordImmutableTimestamp(immutableTsResponse.getImmutableTimestamp());
+            Supplier<Long> startTimestampSupplier = getStartTimestampSupplier();
+
+            AdvisoryLockPreCommitCheck advisoryLockCheck =
+                    AdvisoryLockPreCommitCheck.forLockServiceLocks(lockTokens, getLockService());
+            SnapshotTransaction transaction = createTransaction(immutableTs, startTimestampSupplier,
+                    immutableTsLock, advisoryLockCheck);
+            return new RawTransaction(transaction, immutableTsLock);
         } catch (Throwable e) {
-            if (lock != null) {
-                lockService.unlock(lock);
-            }
+            timelockService.unlock(ImmutableSet.of(immutableTsResponse.getLock()));
             throw Throwables.rewrapAndThrowUncheckedException(e);
```

Comment:
yeah this is fair, though would rather punt on this for this PR given how many changes there are already